watertap/core/zero_order_sido.py

```diff
@@ -359,32 +312,67 @@ def calculate_scaling_factors(self):
-                    default=1e-2,
-                    warning=True,
-                    hint=" for treated temperature equality"))
-
-        for t, v in self.byproduct_temperature_equality.items():
-            iscale.constraint_scaling_transform(
-                v, iscale.get_scaling_factor(
-                    self.properties_in[t].temperature,
-                    default=1e-2,
-                    warning=False))  # would just be a duplicate of above
+    def load_parameters_from_database(self):
+        raise NotImplementedError()
+
+    def set_param_from_data(
+            self, parameter, data, index=None, use_default_removal=False):
+        pname = parameter.parent_component().local_name
+
+        try:
+            pdata = data[pname]
+        except KeyError:
+            raise KeyError(
+                f"{self.name} - database provided does not contain an entry "
+                f"for {pname} for technology.")
+
+        if index is not None:
+            try:
+                pdata = pdata[index]
+            except KeyError:
+                if pname == "removal_mass_solute" and use_default_removal:
+                    try:
+                        pdata = data["default_removal_mass_solute"]
+                        index = "default"
+                    except KeyError:
+                        raise KeyError(
+                            f"{self.name} - database provided does not "
+                            f"contain an entry for {pname} with index {index} "
+                            f"for technology and no default removal was "
+                            f"specified.")
+                else:
+                    raise KeyError(
+                        f"{self.name} - database provided does not contain "
+                        f"an entry for {pname} with index {index} for "
+                        f"technology.")
+
+        try:
+            val = pdata["value"]
+        except KeyError:
+            raise KeyError(
+                f"{self.name} - no value provided for {pname} (index: "
+                f"{index}) in database.")
+        try:
+            units = getattr(pyunits, pdata["units"])
+        except KeyError:
+            raise KeyError(
+                f"{self.name} - no units provided for {pname} (index: "
+                f"{index}) in database.")
+
+        parameter.fix(val*units)
+        _log.info_high(f"{parameter.name} fixed to value {val} {str(units)}")
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {"Inlet": self.inlet,
+             "Treated": self.treated,
+             "Byproduct": self.byproduct},
+            time_point=time_point)
+
+    def _get_performance_contents(self, time_point=0):
+        var_dict = {"Water Recovery": self.recovery_vol[time_point]}
+        for (t, j), v in self.removal_mass_solute.items():
+            if t == time_point:
+                var_dict[f"Solute Removal [{j}]"] = v
```

Comment:
Alternatively, using slices:
```suggestion
        for j, v in self.removal_mass_solute[time_point,:].wildcard_items():
            var_dict[f"Solute Removal [{j}]"] = v
```